crates/iceberg-catalog/src/service/task_queue/mod.rs

```diff
@@ -0,0 +1,85 @@
+use async_trait::async_trait;
+use chrono::Utc;
+use sqlx::FromRow;
+use std::time::Duration;
+use uuid::Uuid;
+
+pub mod delete_queue;
+pub mod tabular_expiration_queue;
+
+#[async_trait]
+pub trait TaskQueue: std::fmt::Debug {
+    type Task: Send + Sync + 'static;
+    type Input: Send + Sync + 'static;
+
+    fn queue_name(&self) -> &'static str;
+    async fn poll(&self) -> crate::api::Result<Option<Self::Task>>;
+    async fn record_success(&self, id: Uuid) -> crate::api::Result<()>;
+    async fn record_failure(
+        &self,
+        id: Uuid,
+        n_retries: i32,
+        error_details: String,
+    ) -> crate::api::Result<()>;
+    async fn enqueue(
+        &self,
+        task: Self::Input,
+        // TODO: taking transaction here enables us to have all-or-nothing semantics for enqueuing
```

Comment:
Would we benefit from all-or-nothing semantic a lot? We already have the idempocy key so enquing something twice is handled.